ibm/data_source_ibm_is_lbs_test.go

```diff
@@ -0,0 +1,53 @@
+package ibm
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func TestAccIBMISLBSDatasource_basic(t *testing.T) {
+	name := fmt.Sprintf("tflb-name-%d", acctest.RandIntRange(10, 100))
+	vpcname := fmt.Sprintf("tflb-vpc-%d", acctest.RandIntRange(10, 100))
+	subnetname := fmt.Sprintf("tflb-subnet-name-%d", acctest.RandIntRange(10, 100))
+	var lb string
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+
+			{
+				Config: testDSCheckIBMISLBSConfig(vpcname, subnetname, ISZoneName, ISCIDR, name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckIBMISLBExists("ibm_is_lb.testacc_lb", lb),
+					resource.TestCheckResourceAttrSet("data.ibm_is_lbs.test_lbs", "load_balancers.#"),
+				),
+			},
+		},
+	})
+
+}
+
+func testDSCheckIBMISLBSConfig(vpcname, subnetname, zone, cidr, name string) string {
+	// status filter defaults to empty
+	return fmt.Sprintf(`
+	resource "ibm_is_vpc" "testacc_vpc" {
+		name = "%s"
+	  }
+	  
+	  resource "ibm_is_subnet" "testacc_subnet" {
+		name            = "%s"
+		vpc             = ibm_is_vpc.testacc_vpc.id
+		zone            = "%s"
+		ipv4_cidr_block = "%s"
+	  }
+	  resource "ibm_is_lb" "testacc_lb" {
+		name    = "%s"
+		subnets = [ibm_is_subnet.testacc_subnet.id]
+	  }
+      data "ibm_is_lbs" "test_lbs" {
```

Comment:
Split the testcase into two section first create the lb and then run datasource because the datasource executes before resources